### ISSUE TITLE
Inconsistency in greenlet dependency in 2.0.28 release causes problems with requirements.txt files containing hashes

### ISSUE THREAD

**Issue by inklesspen** [sqlalchemy/sqlalchemy#11129]
### Describe the bug

In the 2.0.28 tag, the greenlet dependency is defined in setup.cfg, as follows:

```
    greenlet != 0.4.17;(platform_machine=='aarch64' or (platform_machine=='ppc64le' or (platform_machine=='x86_64' or (platform_machine=='amd64' or (platform_machine=='AMD64' or (platform_machine=='win32' or platform_machine=='WIN32'))))))
```

On my Macbook Pro M3 Pro, `platform.machine()` returns `'arm64'` rather than `'aarch64'`, with the result that the dependency doesn't exist there. This is probably because of the [installation note](https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#asyncio-platform-installation-notes-including-apple-m1) about Apple Silicon platforms.

Unfortunately, this means that a requirements.txt file with hashes (such as produced by pip-compile with --generate-hashes enabled) that is created on an Apple Silicon mac will be missing a dependency needed by the latest SQLAlchemy release to install on pretty much any other major platform; I was very surprised when I couldn't install packages from my requirements.txt file on my x86_64 Linux server.

I note that in the main branch, this isn't likely to be an issue anymore because 765ac5fc959eb05a5c80fef7a50cdf00d530e416 changed how this dependency was specified, but it is still an issue on the latest release (from five days ago).

### SQLAlchemy Version in Use

2.0.28

### DBAPI (i.e. the database driver)

N/A

### Database Vendor and Major Version

N/A

### Python Version

3.12.2

### Operating system

Mac, Linux

### To Reproduce

```python
1. On an Apple Silicon Mac, use `pip-compile --generate-hashes` to generate a requirements.txt file including a SQLAlchemy dependency.
2. On a x86_64 Linux machine, attempt to install dependencies from the requirements.txt file.
```

### Error

```
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    greenlet!=0.4.17 from https://files.pythonhosted.org/packages/bd/37/56b0da468a85e7704f3b2bc045015301bdf4be2184a44868c71f6dca6fe2/greenlet-3.0.3-cp312-cp312-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl (from sqlalchemy==2.0.28->-r /home/rose/mypkg/requirements.txt (line 173))
```

**Comment by CaselIT** [comment]
Hi,

Use the asyncio extra depending of you are using async features of sqlalchemy.
That's how it will work on 2.1

**Comment by inklesspen** [comment]
I'm **not using** asyncio.

The problem is that version 2.0.28 _always_ requires greenlet on Linux, but not on Mac OS (Apple Silicon), because the dependency definition is incorrect. This means that the requirements.txt I generate on Mac OS cannot successfully install the dependencies on Linux.

**Comment by zzzeek** [comment]
can you not add a requirement with a platform qualifier for now?     this issue has been fixed for 2.1 but we can't change anything now
